### PR TITLE
BitcoinNetwork: simplify/standardize and canonicalize lowercase for user-facing

### PR DESCRIPTION
**Pull request by msgilligan**
This PR corrects some issues with b375a96565aa4c51587451cb7dea2654f37d5425 and brings things more in alignment with Java conventions (see _Effective Java (Third Edition)_, **Item 34: Use enums instead of int constants** -- especially the discussion of `valueOf(String)`, `toString()` and `fromString(String)` on p. 164)

Specifically:
* Add an override of `toString()` to output the string in canonical, lower-case format.
* Remove the `.of(String)` method from b375a96565aa4c51587451cb7dea2654f37d5425 as it duplicates the built-in `.valueOf()`
* Remove the `ofId(String)` method from b375a96565aa4c51587451cb7dea2654f37d5425 in favor of the using the `fromIdString(String)` method with `orElseThrow()` or similar if throwing an exception is desired.
* Rename `.find(String)` to `.fromString(String)` to use the suggested name from _Effective Java_.
* Rename `.findById(String)` to `.fromIdString(String)` for consistency.
* Improve the JavaDoc.

In summary, this reduces the number of static factory methods from 5 to 3, conforms better with Java conventions, and clarifies our preference for lowercase strings as canonical for user-facing input/output.

### CODE REVIEW

**Comment by msgilligan** [comment]
> Looks good. Do you think you could add some tests for those 3 methods?

Yes. Will do.

**Comment by msgilligan** [comment]
@schildbach Tests added.

**Comment by schildbach** [comment]
Merged.
